### PR TITLE
Update to Controller & Database Methods

### PR DESCRIPTION
**Pull request by daniel-fitzg**
Added new methods to the Controller class.
- Register employee
- Update employee details
- Get employee details
- Delete employee
Other methods may be added later.

The code for DataStore class has been included.

Controller tests for manager-specific branches only have been added.

### CODE REVIEW

File: src/test/java/EmployeeControllerTest.java [hunk 1]
```diff
@@ -16,12 +19,167 @@ static void printComplete() {
...
+
+        // Create new Director
+        Director director = new Director(employeeId, "Mary Flynn", "Kerry", "21/04/1978",
+                "4858697A", 120000, "Dublin");
+        employeeController.registerEmployee(director, true);
+
+        // Update address of Director
+        director = new Director(employeeId, "Mary Flynn", newEmployeeAddress, "21/04/1978",
+                "4858697A", 120000, "Dublin");
+        employeeController.updateEmployeeDetails(director, true);
```
Comment (rdmoraes) [inline src/test/java/EmployeeControllerTest.java:68]: Just wondering if there is a set method to change employees address or have to overload the Director object in every single modification in employees details. This is just an observation no need changes in the code!

**Review by rdmoraes — APPROVED** [review]
WoW this was a big review!! Anyway has my approval

File: src/test/java/EmployeeControllerTest.java (hunk 1 above)
Comment (daniel-fitzg) [reply src/test/java/EmployeeControllerTest.java:68]: Screenshot 2020-04-28 at 12 54 40

In abstract Employee class there are fields and methods that are concrete. So any employee object (Director, DaparmentManager, Staff etc..) can set any of the concrete fields in the abstract Employee superclass.

Comment (daniel-fitzg) [reply src/test/java/EmployeeControllerTest.java:68]: This is the confusing part about abstract classes.

Comment (rdmoraes) [reply src/test/java/EmployeeControllerTest.java:68]: Ok thanks Danny!

**Pull request merged**